### PR TITLE
Utilities: Implement `cmp(1)`

### PR DESCRIPTION
**Pull request by BertalanD**
This PR implements the [`cmp` utility](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/cmp.html), which is needed if we want to run bash's `configure` script inside Serenity.

---
**Utilities: Implement cmp(1)**
This POSIX utility tells us whether the contents of two files are the
same. It's used by bash's configure script.

### CODE REVIEW

File: Userland/Utilities/cmp.cpp [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+}
+
+static ErrorOr<FILE*> open_fd(int fd)
+{
+    if (fd == STDIN_FILENO)
+        return stdin;
+    auto* maybe_fp = fdopen(fd, "r");
+    if (maybe_fp == nullptr)
+        return Error::from_errno(errno);
+    return maybe_fp;
```
Comment (linusg) [inline Userland/Utilities/cmp.cpp:40]: Can we have a Core::System wrapper for this instead?

**Comment by awesomekling** [comment]
What's the rationale behind using old-style C stdio APIs instead of our own C++ libraries here?

**Comment by BertalanD** [comment]
This seemed like the straightforward implementation. I could probably replace the non-optimized code path with `Core::Stream::BufferedFile::read`s into single-byte buffers, but I'm not exactly sure if those would make this code any prettier.

**Comment by bgianfo** [comment]
It seems very weird to write a new app to the C APIs when we have the explicit goal of writing everything in modern C++. I don't see why this utility is special in any way?

**Comment by BertalanD** [comment]
I don't have the time/motivation to fight with and figure out `Core::Stream`.

The bash port is probably going to work without this, as the shell error from the missing `/bin/cmp` is going to be treated as if the compared files are different, causing the needed files to be (re)generated :man_shrugging:  .

**Pull request closed without merging**